### DISCUSSION TITLE
Vanilla features that are not in FA (yet)

### DISCUSSION THREAD

**Discussion by LevFendi** [Factorio-Access/FactorioAccess#187] [General]
This list may take a while to complete and many of these items are already on the issues list but let's try to summarize here...

- Achievements (there is api access)

- Setting train limits at train stops

- Setting complex train schedules from the train menu

- Train stops interfacing with the circuit network

- Extra circuit network features for roboports and train signals and a couple other things (check the wiki page)

- Geometrically arranging the armor equipment grid (it would be like 25% more efficient).

- managing gun and ammo slots like inventory slots (because they are tiny inventories)

- Detailed production info graphs for items, fluids, electricity.

- A tech tree that shows which technology leads to which technologies next.

- The ability to set any logistic request with an item selector similar to the signal selector.

- Quantitative tool tip info with dynamic values AND base values. Stuff like pollution and power consumption data.

- Vanilla factorio's extra scenarios and minigames (rocket rush, transport belt madness, etc.)

- A smooth multiplayer experience

**Comment by EphDoering** [comment]
* selecting, reading, and clearing filters on supported inventory slots
* reading, input, output, or current circuit network signals
* modded guis
* read things that got printed with some variant of print
* read the current goal (used by tutorial, scenarios, and mods)
* pausing while looking at tech tree in single player
* chart tags
* multiplayer chart pings
* previewing details of each unlock of a technology
* train overview menu
* bonus overview menu
* modded selection tools (like creep shovel in KS2)
* rail planner
* configurations for selection tools (like deconstruction planner, and upgrade planner)
* blueprint library
* mod settings (probably a combo of launcher and in mod to cover everything)
* regular settings (mostly, if not all, launcher's responsibility)
* select to save replay data or "view" a replay
* ensure electricity coverage while dragging power poles
* auto underground with dragging belts
* connection previews (for example an inserter will highlight the machines that it's taking from or placing to) and that's including with blueprint placement
* map? there's a lot of options for different visibility layers that each can convey a lot of useful information, but I think each layer probably has a different best way to present the info (like I use the robot layer to look for bots that are stuck in recharge loops, or that I lost while I'm out and about, but that info would be impossible to glean from just reading out a bunch of bot positions)
* logistics overview

some of these are giant with many sub features, and some have api related blockers.

**Comment by ahicks92** [comment]
We will probably not get pausing for anything because the mod pauses with it.

We have current circuit network signals.  We don't have decider or arithmetic combinator.  Those are prereqs to needing to read input different from output.

Blueprint library is probably a no unless we make our own.

We don't have copy/paste history. Coming in Factorio 2.0 provided we can be compatible enough to make it worthwhile to play.

We should list what the map layers are.  As the blind dev it's great to know they're present and I'm sure they're very useful but...

I don't believe we're ever going to have a smooth multiplayer experience.  That should be tracked and given more success criteria.  In particular it is not possible for us to get around the lag, and unless we can you have to be internetically close to the server.  What I have considered is if one of us wants to run servers that reset once a week.  It's like $30 to get something up on gcp but most blind people cannot afford to do that.  We should find out what solutions might already exist to help with the networking.  I have seen products for sighted games which say they do this but don't at the moment remember what they were.  I know from work that proxies via an internal cloud's networking can help the issue.  Of course for sighted players none of this is needed but if we did get something working and some docker images to deploy it, people in general would love us.

**Comment by LevFendi** [comment]
This discussion is VERY outdated. If we want to create such a list again, it should be done from scratch. Luckily it would also be way shorter.
